### ISSUE TITLE
calico-kube-controllers with K8S_INSECURE_SKIP_TLS_VERIFY

### ISSUE THREAD

**Issue by christiancadieux** [projectcalico/calico#7191]
on 3.24.5

I added a configmap to configure eBPF without kube-proxy :
```
kind: ConfigMap
apiVersion: v1
metadata:
  name: kubernetes-services-endpoint
  namespace: tigera-operator
data:
  KUBERNETES_SERVICE_HOST: "..."
  KUBERNETES_SERVICE_PORT: "6443"
```
then I get this error:
```
2023-01-14 21:08:57.452 [INFO][6] main.go 134: Failed to initialize datastore error=Get "https://anvil2-vpis-sit-01....net:6443/apis/crd.projectcalico.org/v1/clusterinformations/default": x509: certificate is valid for localhost, kubernetes, kubernetes.default, kubernetes.default.svc, kubernetes.default.svc.cluster.local, *, not anvil2-vpis-sit-01.....net
```

then, if I add ENV K8S_INSECURE_SKIP_TLS_VERIFY = true to calico-kube-controllers deployment to avoid this error:

the pod refuse to start:
```
2023-01-14 21:09:17.362 [FATAL][6] main.go 116: Failed to start error=failed to build Calico client: invalid configuration: default cluster has no server defined
```

## Expected Behavior
should work

## Current Behavior
pod fails

## Context
need to configure eBPF without kube-proxy

## Your Environment
3.24.5

**Comment by caseydavenport** [comment]
You shouldn't use `K8S_INSECURE_SKIP_TLS_VERIFY`

The original error suggests that your TLS configuration for the kube-apiserver isn't correct. Your certificates need to include the hostname that you will use for your LB.

**Comment by christiancadieux** [comment]
what if there are hundreds  of clusters , they are not on the internet , there is no plan on getting signed certificates for each of them and access to these existing clusters is already only done using skip_tls_verify

**Comment by tomastigera** [comment]
IS that just a bpf issue? This seems like a generic issue you would have.

**Comment by caseydavenport** [comment]
I would still suggest not using TLS verification. Obviously it's up to you, but it's bad security practice.

Personally, I would go so far as suggesting that we remove this option from our code altogether. You can generate self-signed certs for each apiserver, and whatever tool you're using to install Kubernetes should support that already.

**Comment by caseydavenport** [comment]
> IS that just a bpf issue? This seems like a generic issue you would have.

It doesn't seem BPF related to me. 

> failed to build Calico client: invalid configuration: default cluster has no server defined

This suggests that the kubeconfig given to the CNI plugin is missing information about where the API server can be found

**Comment by christiancadieux** [comment]
it started when I tried to install eBPF, but I agree it's not an eBPF problem. installing eBPF requires that a configmap: kubernetes-services-endpoint be created with the 'real' address of the kube-api  to remove the dependency on kub-proxy.
in our case:
This address does not have a valid certificate so it can only be used with insecure-skip-tls-verify. But each node does have a valid kubeconfig with the address 'kubernetes-public-vip' . that worked fine for the tigera-operator because that operator runs in 
```
dnsPolicy: ClusterFirstWithHostNet
hostNetwork: true
```
So, with that new configmap created with 'kubernetes-public-vip', the operator works but now all the other services start failing because they also start using 'kubernetes-public-vip' but these other services are not running  with 'hostNetwork:true' . 
So yes, we should create valid certificates for  the clusters but that's not going to happen (also I should not run with scissors but sometimes I do). 
calico and calico-enterprise was working just fine until I tried to create that configmap, so calico does not care that much whether we have valid certs or not. 
An insecure-skip-tls-verify flag that works , just like the one available in kubeconfig would be very nice, so I can continue to evaluate and maybe buy calico-enterprise.
maybe an extra "slight hack" in your code, around https://github.com/tigera/operator/blob/master/main.go#L85 :-)
